### DISCUSSION TITLE
Need parent dir for computeJacobian under phase_field

### DISCUSSION THREAD

**Discussion by Dadakskn** [idaholab/moose#24977] [Q&A Modules: General]
hi,

I am currently working on implementing the Multi-scale Finite Volume (MsFV) method into the MOOSE framework. As part of this implementation, I require the parent directory for the computeJacobian method, which is responsible for generating a Jacobian matrix within the phase_field module. However, I have been unable to locate the specific directory, and I would greatly appreciate your assistance in this matter.

Thank you very much for your help.

**Comment by GiudGiud** [comment]
Hello

Every kernel and BC should have its own computeJacobian routine
The ones that do not usually leverage automatic differentiation, which is able to compute derivatives in computeResidual calls

Guillaume
